### PR TITLE
implement list of task

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

File: src/main/java/core/basesyntax/Circle.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private int radius;
+
+    public Circle(Colors color, int uniLength) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Circle.java:6]:
```suggestion
    public Circle(Colors color, int radius) {
```

File: src/main/java/core/basesyntax/FigureProducer.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureProducer {
+
+    private ColorsProducer colorsProducer;
+
+    private int uniLength = new Random().nextInt(100);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/FigureProducer.java:9]: make this field constant

File: src/main/java/core/basesyntax/Square.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private int sideLength;
+
+    public Square(Colors color, int uniLength) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Square.java:6]:
```suggestion
    public Square(Colors color, int sideLength) {
```
same with other constructors

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
nice structure, left some minore comments

File: src/main/java/core/basesyntax/Circle.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/core/basesyntax/Circle.java]: done

File: src/main/java/core/basesyntax/Square.java (hunk 3 above)
Comment (RuslanIbIsh) [reply src/main/java/core/basesyntax/Square.java]: done

File: src/main/java/core/basesyntax/FigureProducer.java (hunk 2 above)
Comment (RuslanIbIsh) [reply src/main/java/core/basesyntax/FigureProducer.java]: done

File: src/main/java/core/basesyntax/Figure.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public abstract class Figure implements Drawable {
+    private String name;
+    private Colors color;
+    private double area;
+
+    public Figure(Colors color) {
+        this.color = color;
+    }
+
+    public abstract void obtainArea();
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Figure.java:12]: create new interface and move this method there

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    public IsoscelesTrapezoid(Colors color, int uniLength) {
+        super(color);
+        this.parallelSide = uniLength;
+    }
+
+    @Override
+    public void obtainArea() {
+        double sideA = parallelSide / 2;
+        double sideB = parallelSide * 1.5;
+        double p = (sideA + sideB + (parallelSide * 2)) / 2;
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/IsoscelesTrapezoid.java:15]: p - is not informative

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]
left two comments, rest is good, great job

File: src/main/java/core/basesyntax/Figure.java (hunk 4 above)
Comment (RuslanIbIsh) [reply src/main/java/core/basesyntax/Figure.java]: done

File: src/main/java/core/basesyntax/IsoscelesTrapezoid.java (hunk 5 above)
Comment (RuslanIbIsh) [reply src/main/java/core/basesyntax/IsoscelesTrapezoid.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**